src/visualization/vega_renderer/JSCanvas.m

```diff
@@ -268,32 +267,41 @@ - (void)setFont:(NSString *)fontStr {
 
     assert(fontSize > 0 && fontSize < 1000);
 
-    NSFont *font;
+    NSFont *newFont;
     if (fontProperties.fontFamily == nil) {
-        font = [NSFont systemFontOfSize:fontSize];
+        newFont = [NSFont systemFontOfSize:fontSize];
     } else {
         NSFontManager *fontManager = [NSFontManager sharedFontManager];
         NSArray<NSString *> *possibleFontFamilies = [fontProperties.fontFamily componentsSeparatedByString:@","];
         for (NSString * __strong possibleFontFamily in possibleFontFamilies) {
             possibleFontFamily = [possibleFontFamily stringByTrimmingCharactersInSet:[NSCharacterSet whitespaceCharacterSet]];
-            font = [NSFont fontWithName:possibleFontFamily size:fontSize];
-            if (font != nil && fontProperties.fontWeight != nil) {
+            newFont = [NSFont fontWithName:possibleFontFamily size:fontSize];
+            if (newFont != nil && fontProperties.fontWeight != nil) {
                 if ([fontProperties.fontWeight isEqualToString:@"bold"]) {
-                    font = [fontManager convertFont:font toHaveTrait:NSBoldFontMask];
-                    assert(font != nil);
+                    newFont = [fontManager convertFont:newFont toHaveTrait:NSBoldFontMask];
+                    assert(newFont != nil);
                 } else {
                     // unexpected font weight
                     assert(false);
                 }
             }
-            if (font != nil) {
+            if (newFont != nil) {
                 break;
             }
         }
     }
-    assert(font != nil);
-    _nsFont = font;
+
+    if(newFont == nil) {
+        newFont = [NSFont systemFontOfSize:fontSize];
+        // TODO should we be updating _font to reflect the system font we've fallen back to?
+        NSLog(@"The specified font: '%@' is unavailable. Falling back to '%@'.", fontStr, [newFont displayName]);
+    } else {
+        _font = fontStr;
```

Comment:
Do we still need to store both `_font` and `_nsFont` on the instance of this class? Or should we get rid of `_font`?

Reply:
I'm not sure how else we would meet the requirements for the canvas `font` property otherwise https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D/font

Reply:
Oh, right. I keep forgetting that's why it's in there.